torch/utils/data/dataloader.py

```diff
@@ -175,7 +175,7 @@ def _pin_memory_loop(in_queue, out_queue, device_id, done_event):
         else:
             idx, batch = r
             try:
-                batch = pin_memory_batch(batch)
+                batch = pin_memory_batch(batch, pin_fn)
```

Comment:
What about replacing the `pin_memory_batch` by `pin_fn`, and by default have it be `pin_memory_batch`?

Reply:
Fine by me.